File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/StarsProgram.java
package com.lumiyaviewer.lumiya.render.shaders;

import android.opengl.GLES20;
import com.lumiyaviewer.lumiya.render.RenderContext;

public class StarsProgram extends ShaderProgram {
    public int uMVPMatrix;
    public int uStarColor;
    public int vPosition;

    public StarsProgram() {
        super(Shader.StarsVertexShader, Shader.StarsFragmentShader);
    }

    public void ApplyWindlight(RenderContext renderContext) {
        GLES20.glUniform4f(this.uStarColor, 1.0f, 1.0f, 1.0f, renderContext.windlightPreset.star_brightness);
    }

    @Override
    public /* bridge */ /* synthetic */ int Compile(ShaderPreprocessor shaderPreprocessor) throws ShaderCompileException {
        return super.Compile(shaderPreprocessor);
    }

    @Override
    protected void bindVariables() {
        this.vPosition = GLES20.glGetAttribLocation(this.handle, "vPosition");
        this.uMVPMatrix = GLES20.glGetUniformLocation(this.handle, "uMVPMatrix");
        this.uStarColor = GLES20.glGetUniformLocation(this.handle, "uStarColor");
    }

    @Override
    public /* bridge */ /* synthetic */ int getHandle() {
        return super.getHandle();
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/shaders/StarsProgram.kt
package com.lumiyaviewer.lumiya.render.shaders

import android.opengl.GLES20
import com.lumiyaviewer.lumiya.render.RenderContext

class StarsProgram : ShaderProgram(Shader.StarsVertexShader, Shader.StarsFragmentShader) {
    @JvmField var uMVPMatrix = 0
    @JvmField var uStarColor = 0
    @JvmField var vPosition = 0

    fun ApplyWindlight(renderContext: RenderContext) {
        GLES20.glUniform4f(uStarColor, 1.0f, 1.0f, 1.0f, renderContext.windlightPreset.star_brightness)
    }

    override fun bindVariables() {
        vPosition = GLES20.glGetAttribLocation(handle, "vPosition")
        uMVPMatrix = GLES20.glGetUniformLocation(handle, "uMVPMatrix")
        uStarColor = GLES20.glGetUniformLocation(handle, "uStarColor")
    }
}
